tests/shapes/test_tpms.py

```diff
@@ -340,7 +313,7 @@ def test_tpms_given_density_must_return_corresponding_offset():
         density=0.5,
         part_type="sheet",
     )
-    assert 0 < offset < 2.0 * np.max(tpms.grid["surface"])
+    assert 0 < offset < -2.0 * np.min(tpms.grid["surface"])
```

Comment:
This line seems too complicated maybe you can store the `-2.0 * np.min(tpms.grid["surface"])` inside a variable with an explicit name ?

Or maybe rename this test, because you are not checking "corresponding" offset given a density, you are just checking that the offset is somehow valid by its range.

Also try starting your test name with 
`test_tpms_offset_from_density_given_density_must...`
as you are testing the static method `offset_from_density`